domains/client/domain-operator/src/snap_sync.rs

```diff
@@ -12,22 +13,107 @@ use sc_network_common::sync::message::{
 use sc_network_sync::block_relay_protocol::BlockDownloader;
 use sc_network_sync::service::network::NetworkServiceHandle;
 use sc_network_sync::SyncingService;
-use sp_api::ProvideRuntimeApi;
 use sp_blockchain::HeaderBackend;
 use sp_consensus::BlockOrigin;
-use sp_core::H256;
 use sp_domains::ExecutionReceiptFor;
-use sp_mmr_primitives::MmrApi;
 use sp_runtime::traits::{Block as BlockT, Header as HeaderT, NumberFor};
 use std::collections::HashSet;
+use std::sync::atomic::{AtomicBool, Ordering};
 use std::sync::Arc;
 use std::time::{Duration, SystemTime, UNIX_EPOCH};
-use subspace_service::domains::ConsensusChainSyncParams;
-use subspace_service::mmr_sync;
-use subspace_service::sync_from_dsn::snap_sync_engine::SnapSyncingEngine;
+use subspace_sync::snap_sync_engine::SnapSyncingEngine;
+use tokio::sync::broadcast;
+use tokio::sync::broadcast::{Receiver, Sender};
```

Comment:
It’s a bit confusing to mix `futures::channel::mpsc` and `tokio::sync::broadcast` channels in the same file. But I understand that it’s probably because they have different behaviour when they get full.

Can you change `Sender` and `Receiver` into `broadcast::Sender` and `broadcast::Receiver` to make the code more readable?